src/App.tsx

```diff
@@ -15,35 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabetically = 'alphabetically',
+  byLength = 'byLength',
+  default = '',
+}
+
+type SortObj = {
+  sortField: SortType;
+  isReversed?: boolean;
+};
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed = false }: SortObj,
+) {
+  const preparedGoods = [...goods];
+
+  preparedGoods.sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.alphabetically:
+        return good1.localeCompare(good2);
+
+      case SortType.byLength:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.default);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibledGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
+  const reset = () => {
+    setIsReversed(false);
+    setSortField(SortType.default);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SortType.alphabetically,
+          })}
+          onClick={() => setSortField(SortType.alphabetically)}
+        >
           Sort alphabetically
```

Comment:
Consider marking the active button to better indicate to the user which sorting action is currently active. This could be done by changing the button's style or using an icon, in line with the [UX] checklist.